src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7155,6 +7155,7 @@ private bool IsPossibleStatement(bool acceptAccessibilityMods)
                 case SyntaxKind.PrivateKeyword:
                 // could be a local function with attributes
                 case SyntaxKind.OpenBracketToken:  // PROTOTYPE(local-function-attributes): reuse this or always allow the OpenBracketToken?
+                case SyntaxKind.ExternKeyword:
```

Comment:
>case SyntaxKind.ExternKeyword: [](start = 16, length = 30)

Why is this tied to an accessibility modifier? I would expect this keyword handled in the same way ```SyntaxKind.StaticKeyword``` is handled. The same question is for ```SyntaxKind.OpenBracketToken```.

Reply:
I found that moving the case OpenBracketToken to unconditionally return 'true' caused the parser to hang in some error scenarios. Looking into why and if there's maybe a fix to make elsewhere in the parser that will rationalize all this.

---
In reply to: [354046434](https://github.com/dotnet/roslyn/pull/40011#discussion_r354046434) [](ancestors = 354046434)

Reply:
The original implementation was concealing a bug in here where attributes on non-local-function statements are being rejected in parsing more or less by accident, when it's expected that they will parse and we will give errors when binding the attributes on disallowed statement kinds. I am adding a prototype comment and punting to the next PR as the change will have some rippling effects in tests.

---
In reply to: [356317803](https://github.com/dotnet/roslyn/pull/40011#discussion_r356317803) [](ancestors = 356317803,354046434)